src/main/java/org/terasology/behaviors/actions/NearbyBlockRestricted.java

```diff
@@ -0,0 +1,95 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+package org.terasology.behaviors.actions;
+
+import com.google.common.collect.Lists;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.behaviors.components.StrayRestrictionComponent;
+import org.terasology.logic.behavior.BehaviorAction;
+import org.terasology.logic.behavior.core.Actor;
+import org.terasology.logic.behavior.core.BaseAction;
+import org.terasology.logic.behavior.core.BehaviorState;
+import org.terasology.math.geom.Rect2i;
+import org.terasology.minion.move.MinionMoveComponent;
+
+import java.util.List;
+import java.util.Random;
+import org.terasology.navgraph.WalkableBlock;
+
+/**
+ * Sets a marketkeeper character's {@link MinionMoveComponent} target to a random nearby block inside the area of
```

Comment:
JavaDoc is still specific for the MR shopkeeper NPC 😉

Reply:
whoops! The javadoc here has been edited to be more generic